### ISSUE TITLE
SA1623 and SA1629 reported for non-exposed members even when documentInternalElements is false

### ISSUE THREAD

**Issue by RobSiklos** [DotNetAnalyzers/StyleCopAnalyzers#3096]
When the `documentInternalElements` option is set to false, SA1623:PropertySummaryDocumentationMustMatchAccessors and SA1629:DocumentationTextMustEndWithAPeriod should never be reported for those non-public members which are exempted from the documentation requirement. 

Repro:
```csharp
internal class FalsePositives
{
    /// <summary>I will generate SA1623 and SA1629</summary>
    public long Foo => 0;

    /// <summary>I will generate SA1629</summary>
    public void M() { }

    /// <summary>No warnings - as expected.</summary>
    public event EventHandler MyEvent;
}

```

**Comment by sharwell** [comment]
📝 These are two different settings. `documentInternalElements` controls whether or not documentation is required. SA1623 and SA1629 relate to the validity of the documentation _when provided_. The current behavior is certainly following the design we expected when we implemented the rules, but leaving the issue open for further discussion.

**Comment by RobSiklos** [comment]
Ok, but it's kind of inconsistent. For instance, let's say I have the following:
```csharp
internal class C {
    /// <param name="arg">blah</param>
    public void M(int arg) { arg.GetHashCode(); }
}
```
SA1629 is generated for the `<param>` tag - the documentation must end with a period. However, if I take out that line completely, SA1611 (ElementParametersMustBeDocumented) does not fire (ref issue #2444). So it seems that some documentation rules are more equal than others when it comes to considering `documentInternalElements`.

From what I can tell, our case is the same as that in issue #2361, which is to say, we don't want to make extra work for developers by forcing that their private/internal documentation is absolutely perfect (ends with a period, proper "Gets or sets" text, etc.). However, we DO want public documentation to be subjected to all these nitpicky rules.
